ЗАДАЧА 1
Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.
а = 1 - 0.95
z = 1.96 для а/2 = 0.025

In [1]:
import numpy as np
import scipy.stats as stats

In [2]:
x = 80
d = 16
n = 256

In [3]:
t_1 = 80 - (1.96*16/np.sqrt(256))
t_1

78.04

In [4]:
t_2 = 80 + (1.96*16/np.sqrt(256))
t_2

81.96

Ответ: [78.04; 81.96]

ЗАДАЧА 2
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [5]:
a = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [6]:
x_1 = np.mean(a)
x_1

6.590000000000001

In [7]:
D1 = np.var(a, ddof = 1)
D1

0.2032222222222223

In [8]:
t1 = stats.t.ppf(0.975, 9)
t1

2.2621571627409915

In [9]:
L = x_1 - t1*np.sqrt(D1/10)
L

6.267515851415713

In [10]:
U = x_1 + t1*np.sqrt(D1/10)
U

6.912484148584288

Ответ: [6.27; 6.91]

ЗАДАЧА 3
Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170
Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175
Используя эти данные построить 95% доверительный интервал для разности среднего роста 
родителей и детей.

In [11]:
a = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
b = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])

In [12]:
len(a)

10

In [13]:
len(b)

10

In [14]:
x_1 = np.mean(a)
x_1

166.2

In [15]:
x_2 = np.mean(b)
x_2

168.1

In [16]:
delta = x_1 - x_2
delta

-1.9000000000000057

In [17]:
D1 = np.var(a, ddof = 1)
D1

91.06666666666666

In [18]:
D2 = np.var(b, ddof = 1)
D2

60.10000000000001

In [19]:
D = (D1 + D2)/2
D

75.58333333333334

In [20]:
SE = np.sqrt(D/10 + D/10)
SE

3.8880157750022915

In [21]:
t = stats.t.ppf(0.975, 18)
t

2.10092204024096

In [22]:
L = delta - t*SE
L

-10.068418034506857

In [23]:
U = delta + t*SE
U

6.268418034506846

In [ ]:
Ответ: [-10.07; 6.27]